jest.config.js

```diff
@@ -31,7 +31,8 @@ const globals = {
   // Timeouts rely on setTimeout which takes 32 bit numbers
   maxTimeout: Math.pow(2, 31) - 1,
   testCmd: process.env.PK_TEST_COMMAND,
-  testPlatform: process.env.PK_TEST_COMMAND_DOCKER,
+  testPlatform: process.env.PK_TEST_PLATFORM ?? process.platform,
+  tmpDir: process.env.PK_TEST_TMP_DIR ?? os.tmpdir(),
```

Comment:
The setting `tmpDir` here doesn't quite make sense to me.

Any code should be using `os.tmpdir()` should behave according to whatever `TMPDIR` is set to.

On Linux and containers, this is usually set to `/tmp`.

And all of our containers have `/tmp` built in.

Now if we want to "share" some data on the filesystem between container code and outside the container (whether that's a different container). Then the right way to do this is to:

1. Not use the `/tmp` for this.
2. Use a new path, make this `/srv`
3. Then mount a special path into `/srv`.

So for example when you are doing `docker run`.

You'd want to use:

```
docker run \
  --env PK_NODE_PATH='/srv' \
  --mount type=bind,src="$XXX",dst='/srv'
```

This is how we do this in the `testnet.polykey.io` task definition. Telling explicitly PK to use a particular node path, while mounting into the node path if we want to persist that data.

What is `$XXX`? This location is currently set to `PK_TEST_DATA_PATH`. This variable seems unnecessary. I think the `scripts/docker-run.sh` is intended to act as a "executable file proxy" so it can be used as a file target to execute. This variable should be passed in from the JavaScript code. It should not be hardcoded in `scripts/docker-run.sh`.

This means `tests/utils/exec.ts` (the location which should contain all the `pk*` functions, should decide where this location should be.

Now because you want to share the same location between the host (potentially a host container, and not host machine), and other containers. Then it should be sufficient to mount a temporary directory from `/tmp/...`.

Reply:
Here's a visualisation:

```
                                    ┌─────────────────────┐
                                    │    PolykeyAgent     │
                                    │                     │
                              ┌─────┼─►/srv               │
                              │     │  PK_NODE_PATH=/srv  │
                              │     └─────────────────────┘
                              │
┌────────────────────────┐    │
│ Host                   │    │
│                        │    │     ┌─────────────────────┐
│                        │    │     │   PolykeyClient     │
│ /tmp                   │    │     │                     │
│ /tmp/polykey-test-XXX──┼────┴─────┼─►/srv               │
│                        │          │  PK_NODE_PATH=/srv  │
└────────────────────────┘          └─────────────────────┘
```

The end result is that we do not change anything about the TMPDIR or `os.tmpdir`, and this variable should not be required.

This means the tests should look very similar as before, and you can use `--node-path` instead of `PK_NODE_PATH` to be more explicit.

Reply:
If you already have a `pk*Target` let's just special case docker then.

Then you can specify the `--mount` option directly and also keep the paths the same.

I don't actually see a huge issue with using `/srv` as the directory node path inside the containers. But it can be set to the same path outside.

Therefore on CI/CD, we would see something like `/build/$CI_PROJECT_PATH/tmp/polykey-test-XXX` being used both inside the `PolykeyAgent` and `PolykeyClient` containers and as well as the same build/host container.

Reply:
```
                                    ┌─────────────────────────┐
                                    │    PolykeyAgent         │
                                    │                         │
                              ┌─────┼─►/tmp/polykey-test-XXX  │
                              │     │  PK_NODE_PATH=""        │
                              │     └─────────────────────────┘
                              │
                              │
┌────────────────────────┐    │
│ Host                   │    │     ┌─────────────────────────┐
│                        │    │     │   PolykeyClient         │
│ /tmp                   │    │     │                         │
│ /tmp/polykey-test-XXX──┼────┴─────┼─►/tmp/polykey-test-XXX  │
│                        │          │  PK_NODE_PATH=""        │
└────────────────────────┘          └─────────────────────────┘
```

The `""` means the same path.

Reply:
Alright so if `scripts/docker-run.sh` can be deleted and the same stuff done inside the `tests/utils/exec.sh`, we remove a couple environment variables.

Note however there is still these global/env variable:

1. `PK_TEST_PLATFORM` - defaults to `os.platform()` can be overridden to `'docker'`.
2. `PK_TEST_COMMAND` - target file to execute (it's supposed to be a shell command actually), in the case of `docker` execution, this could be set to the path to `docker` itself...
3. `PK_TEST_DATA_PATH` - the data path to be used by the tests, right now all the tests are using `os.tmpdir()`, it should default to using `PK_TEST_DATA_PATH`, which itself defaults to `os.tmpdir()`, on the CI/CD this is set to under the `/build/$CI_PROJECT_PATH` because only this location shareable between the the dind containers

Reply:
Try and see what happens if you were to allow `shell` execution for `PK_TEST_COMMAND`. I believe we normally use `execFile`, but it can be changed to `exec`. Some things may not work. This will end up allowing arbitrary shell commands.

Reply:
Ok `PK_TEST_COMMAND` is ignored when `PK_TEST_PLATFORM='docker'`.

Reply:
These changes have been made. Mainly the `docker-run.sh` has been removed and inlined within the `pkXTarget` functions.